apps/cowswap-frontend/src/common/hooks/useApproveState.ts

```diff
@@ -32,6 +33,10 @@ export function useApproveState(amountToApprove: Nullish<CurrencyAmount<Currency
       return ApprovalState.UNKNOWN
     }
 
+    if (FractionUtils.gte(currentAllowance, amountToApprove)) {
```

Comment:
Now we check allowance sufficiency before checking pendingApproval.
So in case if there is a pending tx but allowance is sufficient - the user will be able to swap.